packages/gestalt/src/Modal.js

```diff
@@ -20,21 +20,15 @@ import modalStyles from './Modal.css';
 
 type Props = {|
   accessibilityModalLabel: string,
+  align?: 'left' | 'center',
   children?: Node,
   closeOnOutsideClick?: boolean,
   footer?: Node,
+  heading?: Node,
   onDismiss: () => void,
   role?: 'alertdialog' | 'dialog',
   size?: 'sm' | 'md' | 'lg' | number,
-  ...
-    | {|
-        heading?: Node,
-      |}
-    | {|
-        heading?: string,
-        subHeading?: string,
-        align?: 'left' | 'center',
-      |},
+  subHeading?: string,
```

Comment:
I think the idea here is to only have the subHeading and align props if the heading is a string specifically. I see Flow is getting confused by Node including string, but is it possible to keep the original idea? Can Node be changed to something that's more specifically a React component so it doesn't include string?

Reply:
Discussed and it looks not possible. This is fine since it's similar but less strict.